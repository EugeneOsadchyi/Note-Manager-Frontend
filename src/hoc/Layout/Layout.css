.Content {
  border-width: 0;
  border-left-width: 2px;
  border-right-width: 2px;
  border-style: solid;
  border-color: black;
  padding: 50px 10px;
  height: 100%;
}
