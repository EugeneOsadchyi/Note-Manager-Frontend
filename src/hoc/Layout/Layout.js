import React from 'react';
import PropTypes from 'prop-types';
import './Layout.css';

import Aux from '../Aux';
import Header from './Header';
import Footer from './Footer';

const Layout = ({ children }) => (
  <Aux>
    <Header title="Manager" />
    <main className="Content">
      {children}
    </main>
    <Footer />
  </Aux>
);

Layout.propTypes = {
  children: PropTypes.element.isRequired,
};

export default Layout;
