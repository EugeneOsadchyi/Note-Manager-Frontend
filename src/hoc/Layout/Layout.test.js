import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Layout from './Layout';

describe('Layout', () => {
  const layout = shallow((
    <Layout>
      <h1>
        Some child component
      </h1>
    </Layout>
  ));

  it('renders properly', () => {
    expect(shallowToJson(layout)).toMatchSnapshot();
  });
});
