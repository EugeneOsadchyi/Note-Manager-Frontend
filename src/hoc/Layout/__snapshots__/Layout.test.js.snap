// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders properly 1`] = `
<Component>
  <Header
    title="Manager"
  />
  <main
    className="Content"
  >
    <h1>
      Some child component
    </h1>
  </main>
  <Footer />
</Component>
`;
